src/Core/Services/Implementations/FeatureRoutedCacheService.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Bit.Core.AdminConsole.AbilitiesCache;
+using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Entities.Provider;
+using Bit.Core.AdminConsole.Models.Data.Provider;
+using Bit.Core.Models.Data.Organizations;
+
+namespace Bit.Core.Services.Implementations;
+
+/// <summary>
+/// A feature-flagged routing service for application caching that bridges the gap between
+/// scoped dependency injection (IFeatureService) and singleton services (cache implementations).
+/// This service allows dynamic routing between IVCurrentInMemoryApplicationCacheService and
+/// IVNextInMemoryApplicationCacheService based on the PM23845_VNextApplicationCache feature flag.
+/// </summary>
+/// <remarks>
+/// This service is necessary because:
+/// - IFeatureService is registered as Scoped in the DI container
+/// - IVNextInMemoryApplicationCacheService and IVCurrentInMemoryApplicationCacheService are registered as Singleton
+/// - We need to evaluate feature flags at request time while maintaining singleton cache behavior
+///
+/// The service acts as a scoped proxy that can access the scoped IFeatureService while
+/// delegating actual cache operations to the appropriate singleton implementation.
+/// </remarks>
+public class FeatureRoutedCacheService(
+    IFeatureService featureService,
+    IVNextInMemoryApplicationCacheService vNextInMemoryApplicationCacheService,
+    IVCurrentInMemoryApplicationCacheService inMemoryApplicationCacheService,
+    IApplicationCacheServiceBusMessaging serviceBusMessaging)
```

Comment:
This message will include the whole review status up to this point, so we can comment back and forth.

I think this should solve our problem with the feature flag. I haven’t had a chance to test it at runtime yet, and I still need to add tests for the IApplicationCacheServiceBusMessaging implementations and update the FeatureRoutedCacheService flag.

@eliykat Please let me know if there’s anything else I should add.